pkg/defaults/defaults.go

```diff
@@ -224,6 +224,10 @@ const (
 	// which are not part of Cilium manged pods.
 	EncryptNode = false
 
+	// NodeEncryptionOptOutLabels contains the label selectors for nodes opting out of
+	// node-to-node encryption
+	NodeEncryptionOptOutLabels = "node-role.kubernetes.io/control-plane"
```

Comment:
Does this default means that the control plane node is by default excluded from node encryption? Sounds like something we may want to have a big warning for in the docs.

Reply:
Yep. The docs say:

> By default, Cilium will disable node-to-node encryption on any node
which has the ``node-role.kubernetes.io/control-plane`` label. This is done to
ensure worker nodes are always able to communicate with the kube-apiserver node,
in order to be able to manage their own encryption key. This ensures worker nodes are
never accidentally locked out from the control plane during bootstrap, or
when their private key changes. The label selector matching the control-plane
nodes is configurable via the ``node-encryption-opt-out-labels`` ConfigMap option.

Do you want as to add a note in the beginning of the page?

Reply:
Oh, I must have missed that yesterday. I would just put it in a Note or "Attention !" box to ensure people are aware that some of their node-to-node traffic is unencrypted by default.

Reply:
Good point! We'll revise the docs here to make this a bit more clear (and also document what exactly it means)

Reply:
I've rewritten the docs: https://github.com/cilium/cilium/commit/42859a45b40e5641f1bce86aef66fad8ba823d10

Hope this makes it a bit more clear why this is needed (and what the consequences are of disabling it)